### PR TITLE
Add linux-cachyos-rt

### PR DESCRIPTION
**Pull request by ventureoo**
Adds a new kernel with RT patches. Wasn't tested very thoroughly, but at least it boots normally.
Notice that the config of this kernel can be synchronized just like all other kernels (linux-rt config is not needed).
 
Compared to other kernels:

- Has 1000Hz by default (matches linux-rt in arch)
- Has Idle ticks by default (matches linux-rt in arch)
- With this kernel, changing the `_preempt` parameter has no effect.
- No other schedulers except CFS are supported (and never will be, but it might make sense to experiment with BORE)
- Doesn't work with `_build_zfs`

~~god, give me the strength to support these patches~~

### CODE REVIEW

**Comment by ptr1337** [comment]
I don't know if we really should merge this.
Actually this requires a additional maintenance and I don't really have the spare time for this anymore.
If anyone is willing to do this - its welcome, but we should keep it proper maintained.

Besides that it seems fine for me. Im not sure if we should include this into LTS Kernels (hardned and lts one)

**Comment by ventureoo** [comment]
> I don't know if we really should merge this. Actually this requires a additional maintenance and I don't really have the spare time for this anymore. If anyone is willing to do this - its welcome, but we should keep it proper maintained.
> 
> Besides that it seems fine for me. Im not sure if we should include this into LTS Kernels (hardned and lts one)

If you want, I can be a maintainer.

As for the additional maintenance, I can say that it all depends on how quickly those or other patches will hit the upstream -> more need to rebase. Basically, apart from constantly updating the patch, nothing is needed, as I managed to avoid enabling CONFIG_EXPERT, otherwise it would have been a real headache.

**Comment by ventureoo** [comment]
\> Besides that it seems fine for me. Im not sure if we should include this into LTS Kernels (hardned and lts one)

I just added a description of it in them to make it easier to make changes together for all kernels. But if you don't need it, I can remove it.

**Comment by ventureoo** [comment]
@sirlucjan @ptr1337 The last commit https://github.com/CachyOS/linux-cachyos/pull/127/commits/a92e68f9f74b40a716eb0a59ac48a848942d1fe4 adds a warning about the incompatibility of the RT kernel with ZFS. We could also add a lincense bypass, but that will probably never be the right solution.

**Comment by sirlucjan** [comment]
It is only right not to add ZFS support to linux-cachyos-rt. We cannot change the ZFS licence or get around this in any other way.

**Comment by ptr1337** [comment]
I dont know if we should force to disable zfs on the RT kernel.
If ZFS users want to do the "workaround" then they can do it with inserting a simple line. The problem is that we are not allowed to package it and provide it in a repo - to a user it does not care at all for private usage.

**Review by ptr1337 — CHANGES_REQUESTED** [review]

**Comment by ventureoo** [comment]
> I dont know if we should force to disable zfs on the RT kernel. If ZFS users want to do the "workaround" then they can do it with inserting a simple line. The problem is that we are not allowed to package it and provide it in a repo - to a user it does not care at all for private usage.

This may be true, but are there so many ZFS users who want to use the RT kernel? I don't think using workarounds is a good idea. But if you want, I will of course remove some of the changes.

File: linux-bore/PKGBUILD [hunk 1]
```diff
@@ -206,7 +209,7 @@ source=(
     "${_patchsource}/all/0001-cachyos-base-all.patch")
 
 # ZFS support
-if [ -n "$_build_zfs" ]; then
+if [ -n "$_build_zfs" ] && [ "$_cpusched" != "rt" ]; then
```
[the remark this answers is not in the document]
Comment (ventureoo) [reply linux-bore/PKGBUILD]: Done by https://github.com/CachyOS/linux-cachyos/pull/127/commits/d3788abbffc6fecd45dc5baec264550c814d1093

File: linux-bore/PKGBUILD [hunk 2]
```diff
@@ -644,7 +655,7 @@ build() {
     cd ${srcdir}/${_srcname}
     make ${BUILD_FLAGS[*]} -j$(nproc) all
 
-    if [ -n "$_build_zfs" ]; then
+    if [ -n "$_build_zfs" ] && [ "$_cpusched" != "rt"]; then
```
[the remark this answers is not in the document]
Comment (ventureoo) [reply linux-bore/PKGBUILD]: Done by https://github.com/CachyOS/linux-cachyos/pull/127/commits/d3788abbffc6fecd45dc5baec264550c814d1093

**Review by ptr1337 — APPROVED** [review]

**Review by sirlucjan — APPROVED** [review]

**Pull request merged**